fast64_internal/mk64/f3d_course_parser.py

```diff
@@ -0,0 +1,338 @@
+import re
+import traceback
+from mathutils import Vector
+from ..f3d.f3d_gbi import F3D
+from ..f3d.f3d_material import RDPSettings
+from ..f3d.f3d_parser import F3DParsedCommands, ParsedMacro, math_eval, parseDLData, parseVertexData
+from ..f3d.f3d_writer import F3DVert
+from ..utility import PluginError, float_from_u16_str, gammaInverseValue, int_from_s16_str, readFile, unpackNormal
+
+
+def courseVertexFormatPatterns():
+    # position, uv, color/normal
+    return [
+        # decomp format
+        "\{\s*"
+        + "\{+([^,\}]*),([^,\}]*),([^,\}]*)\}\s*,\s*"
+        + "\{([^,\}]*),([^,\}]*)\}\s*,\s*"
+        + "\{MACRO_COLOR_FLAG\(([^,\}]*),([^,\}]*),([^,\}]*),([^,\}])*\),([^,\}]*)\}\s*"
+        + "\}",
+    ][0]
```

Comment:
```suggestion
    return (
        # decomp format
        r"\{\s*"
        r"\{+([^,\}]*),([^,\}]*),([^,\}]*)\}\s*,\s*"
        r"\{([^,\}]*),([^,\}]*)\}\s*,\s*"
        r"\{MACRO_COLOR_FLAG\(([^,\}]*),([^,\}]*),([^,\}]*),([^,\}])*\),([^,\}]*)\}\s*"
        r"\}"
    )
```
this uses
- just wrap an expression in parentheses to write it over several lines
- raw strings r"", making `\` not do escape sequences (also IDEs often do regex syntax highlighting in those since that's the primary use case)
- just chaining strings like "" "" concatenates the strings together, as in C

Reply:
ok ok I just follow vertexFormatPatterns of F3DContext who don't have regex